### PR TITLE
FRDQ-124-1

### PR DESCRIPTION
**Pull request by hayuna**
# Steps to reproduce
- update dependencies (npm i)
- run app (npm start)
- put `/multistepform` in the URL address
- after submit form, you should see field values in the console

# To Do
- [ ] Improve styling to fix with Figma
- [ ] Add types instead of any
- [x] Fix SVGs
- [ ] Refactor code

### CODE REVIEW

File: src/components/molecules/MultiStepForm/MultiStepForm.tsx [hunk 1]
```diff
@@ -1,25 +1,67 @@
...
-import { IMultiStepFrom } from './MutliStepFrom.model';
+import { useFormProgress } from 'hooks';
+import { RootState } from 'redux/store';
+import { MultistepformNext, MultistepformSubmit } from 'assets';
+import Button from 'components/atoms/Button';
 import Steps from './Steps/Steps';
+import { IMultiStepFrom } from './MultiStepFrom.model';
+import { StyledAction, StyledContent, StyledMultiStepForm } from './MultiStepForm.styled';
+
+const MultiStepForm = ({ onSubmit, children: steps, store }: IMultiStepFrom): JSX.Element => {
```
Comment (olgierdo) [inline src/components/molecules/MultiStepForm/MultiStepForm.tsx:12]: A bit too complicated validation... + redux I recommend this way --> https://codesandbox.io/s/react-hook-form-wizard-form-13yud?file=/src/Step1.js and validation scheme.

Comment (hayuna) [reply src/components/molecules/MultiStepForm/MultiStepForm.tsx:12]: I exactly used this code before and there was some issues to use it inside our project

File: src/components/atoms/Button/Button.styled.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ const BaseButton = styled.button<{ iconPos: IconPosition }>`
   height: 52px;
   border: none;
   border-radius: 6px;
-  ${typography.body.bold.L}
+  ${typography.body.bold.L};
   font-weight: 600;
```
Comment (Krysik) [inline src/components/atoms/Button/Button.styled.ts:14]: You can remove this declaration. You have font-weight from ```${typography.body.bold.L}```.

File: src/components/forms/ExampleForm/steps/Step1.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import reducer from 'redux/reducers/exampleForm';
+import { RootState } from 'redux/store';
+import * as yup from 'yup';
+
+export const Step1 = (): JSX.Element => {
+  const state = useSelector((s: RootState) => s.exampleForm);
+  const dispatch = useDispatch();
+  const { setSurname, setName } = reducer.actions;
+
+  const validate = async (e: any) => {
```
Comment (Krysik) [inline src/components/forms/ExampleForm/steps/Step1.tsx:14]: change any type

File: src/components/molecules/MultiStepForm/MultiStepForm.styled.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import styled from 'styled-components';
+import colors from '../../../styles/colors';
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/MultiStepForm.styled.ts:2]: change to ```styles/colors```

File: src/components/molecules/MultiStepForm/MultiStepForm.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  return (
+    <StyledMultiStepForm>
+      <div>
+        <Steps numberOfSteps={steps.length} currentStep={currentStep} changeStep={goTo} />
+      </div>
+      <StyledContent>{steps[currentStep]}</StyledContent>
+      <StyledAction>
+        {!isFirst && (
+          <Button variant="transparent" onClick={goBack}>
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/MultiStepForm.tsx:35]: This button has wrong font color. You should change it to ```colors.background.Neutral40```

File: src/components/molecules/MultiStepForm/MultiStepForm.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            Back
+          </Button>
+        )}
+        {isError ? (
+          <Button
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/MultiStepForm.tsx:40]: Even if form is valid the color of button icon doesn't change

File: src/components/molecules/MultiStepForm/MultiStepFrom.model.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+export interface IMultiStepFrom {
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/MultiStepFrom.model.ts:3]: I think You have typo in this file and also in filename.

File: src/components/molecules/MultiStepForm/MultiStepForm.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+            icon={isLast ? MultistepformSubmit : MultistepformNext}
+            onClick={(e: any) => {
+              e.preventDefault();
+
+              if (isLast) {
+                handleSubmit();
+              } else {
+                goForward();
+              }
+            }}
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/MultiStepForm.tsx:60]: Move it to separate function

File: src/components/molecules/MultiStepForm/Steps/Step.styled.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  display: flex;
+  justify-content: center;
+  width: 40px;
+  height: 40px;
+  border-radius: 50%;
+  background: ${({ done }) => (done ? colors.background.Primary40 : colors.background.Neutral80)};
+  color: ${({ done }) => (done ? colors.background.NeutralWhite : colors.background.Neutral20)};
+  font-size: 20px;
+  font-weight: bold;
+  line-height: 40px;
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/Steps/Step.styled.ts:30]: In figma I see there should be ```typography.body.bold.L```

File: src/components/molecules/MultiStepForm/Steps/Steps.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+const Steps = ({ numberOfSteps, currentStep, changeStep }: IStep): JSX.Element => {
+  const selectStep = (step: number) => {
+    if (step <= currentStep) {
+      changeStep(step);
+    }
+  };
+
+  return (
+    <StyledWrapper>
+      {[...Array(numberOfSteps)].map((obj, step) => (
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/Steps/Steps.tsx:14]: You could prefix obj with underscore

File: src/components/molecules/MultiStepForm/Steps/Step.styled.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+export const StyledWrapper = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+export const StyledContainer = styled.div`
+  display: flex;
+  align-items: center;
+`;
```
Comment (Krysik) [inline src/components/molecules/MultiStepForm/Steps/Step.styled.ts:12]: You have two components with exactly same styles

**Review by Krysik — CHANGES_REQUESTED** [review]

**Pull request closed without merging**